### PR TITLE
Stop adopting masters with machine API

### PR DESCRIPTION
**Pull request by enxebre**
Master machine assets were introduced by https://github.com/openshift/installer/pull/491 to adopt instances generated by tf. There was never a follow up to leverage adoption for fatrther addon features. Unless there's a good reason to keep the adoption, this PR removes the machine API master awareness and assets generation until there's a more complete story for it

### CODE REVIEW

**Comment by enxebre** [comment]
cc @derekwaynecarr

**Comment by dhellmann** [comment]
Please don't do this. We're going to need the master machines in order to manage those hosts for RHHI.Next.

**Comment by smarterclayton** [comment]
Yeah, even on AWS there are expectations we can reduce manual effort for admins in some scenarios via this.

**Comment by vikaschoudhary16** [comment]
https://github.com/openshift/origin/pull/22698

**Comment by abhinavdahiya** [comment]
This is a no-go. 

Can you explain what
adding features we are not using?
Where does the adoption create problems for machine-api-? What special things do you have to do ie master awareness?

**Comment by enxebre** [comment]
> Please don't do this. We're going to need the master machines in order to manage those hosts for RHHI.Next.
@dhellmann 
The masters machines today are just adopted by the API. The only benefit you get from this today, is you can delete them via cli. They are not a set, you can not scale them in/out. Could you elaborate your user case?

> Yeah, even on AWS there are expectations we can reduce manual effort for admins in some scenarios via this.
@smarterclayton 
Afaik all this adoption gives you atm is an easy way to delete masters via cli/api. I'd argue that's a +1.
Can you elaborate in which scenarios can be beneficial?

> Can you explain what
> addon features we are not using?
> Where does the adoption create problems for machine-api-? What special things do you have to do ie master awareness?
@abhinavdahiya 
All this adoption gives us today is exposing en easier way to delete master machines. Adopting with API machines could be the foundation for higher level addon features to have a robust API driven control plane which could allow things like scaling capabilities for the control plane via sets, autoscaling, stronger integration with etcd, recovering from distaster scenarios. But afaik none of this is in place today.

So may be I'm just missing context on the uses cases pointed here but until those features are in place is not clear to me the benefit from adopting masters atm.

**Comment by smarterclayton** [comment]
There's enough risks to changing that we need a good reason to remove.
What is the good reason to remove?

File: pkg/asset/machines/master.go [hunk 1]
```diff
@@ -226,12 +225,6 @@ func (m *Master) Load(f asset.FileFetcher) (found bool, err error) {
 		return true, err
 	}
 
-	fileList, err := f.FetchByPattern(filepath.Join(directory, masterMachineFileNamePattern))
-	if err != nil {
-		return true, err
-	}
-
-	m.MachineFiles = fileList
```
Comment (staebler) [inline pkg/asset/machines/master.go]: This will break loading the asset from disk. The `MachineFiles` will be empty when the asset is loaded from disk.

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya
> All this adoption gives us today is exposing en easier way to delete master machines. Adopting with API machines could be the foundation for higher level addon features to have a robust API driven control plane which could allow things like scaling capabilities for the control plane via sets, autoscaling, stronger integration with etcd, recovering from distaster scenarios. But afaik none of this is in place today.

It gives cluster-machine-approver a consistent way to approve server certs for kubelets on IPI
It gives teams working on DR scenarios to get their control plane hosts recreated
It gives teams working on DR auto load-balancer re-wiring

control plane -> machines
compute -> machinesets is a great way to give users info that control plane today is pet, when compute is cattle in Openshift 4.

> So may be I'm just missing context on the uses cases pointed here but until those features are in place is not clear to me the benefit from adopting masters atm.

So I do not agree with your resononing that control plane machines objects only serve "delete from CLI", it is an important API layer we build on.

**Comment by dhellmann** [comment]
> > Please don't do this. We're going to need the master machines in order to manage those hosts for RHHI.Next.
> 
> @dhellmann
> The masters machines today are just adopted by the API. The only benefit you get from this today, is you can delete them via cli. They are not a set, you can not scale them in/out. Could you elaborate your user case?

The RHHI.Next case is a baremetal hyperconverged cluster with the provisioning tool for expanding the cluster self-hosted within the cluster. The source of truth about the [nodes, machines, and hosts](https://github.com/metal3-io/metal3-docs/blob/master/design/nodes-machines-and-hosts.md) in the cluster is custom resources in the kubernetes database, and the cluster-api provider we're building accesses that data using the kubernetes API.

Because this is a self-hosted baremetal setup, there is no external component to do hardware monitoring, so we are also building that so we can report to the user about what hardware they have and whether it is healthy. In order to monitor every member of the cluster, including the masters, we need the installer to register them based on information collected from the user at  the time the cluster is built. There is no other API or database for us to query later to "adopt" them.

This sprint we are focusing on finishing the work to migrate the logic from our proof-of-concept scripts for doing all of this into the installer, so we will have more patches soon to show exactly how we will use the Machine objects being registered here as masters, as well as any that will be registered as workers.

**Comment by enxebre** [comment]
@dhellmann @abhinavdahiya thanks for elaborating. As far as we are clear and explicit on the message that masters machines are pets today with the risks which that carries I'm ok.

**Pull request closed without merging**